### PR TITLE
Collapse the list and watch resource version parse

### PR DESCRIPTION
**Pull request by fisherxu**
**What this PR does / why we need it**:
Collapse the list and watch resource version parse, as discuss in [#64513](https://github.com/kubernetes/kubernetes/pull/64513#issuecomment-399380988)
**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/storage/interfaces.go [hunk 1]
```diff
@@ -50,15 +50,11 @@ type Versioner interface {
 	ObjectResourceVersion(obj runtime.Object) (uint64, error)
 
 	// ParseWatchResourceVersion takes a resource version argument and
```
Comment (wgliang) [inline staging/src/k8s.io/apiserver/pkg/storage/interfaces.go]: ParseWatchResourceVersion-> ParseResourceVersion

Comment (fisherxu) [reply staging/src/k8s.io/apiserver/pkg/storage/interfaces.go]: Addressed :).

**Comment by liggitt** [comment]
@kubernetes/sig-api-machinery-pr-reviews 

/lgtm

**Pull request merged**